qualtran/bloqs/factoring/rsa/rsa_phase_estimate.py

```diff
@@ -0,0 +1,141 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import math
+import random
+from functools import cached_property
+from typing import Dict, Optional
+
+import attrs
+import sympy
+from attrs import frozen
+
+from qualtran import (
+    Bloq,
+    bloq_example,
+    BloqBuilder,
+    BloqDocSpec,
+    DecomposeTypeError,
+    QUInt,
+    Signature,
+    SoquetT,
+)
+from qualtran.bloqs.basic_gates import IntState, PlusState
+from qualtran.bloqs.bookkeeping import Free
+from qualtran.bloqs.factoring._factoring_shims import MeasureQFT
+from qualtran.bloqs.mod_arithmetic.mod_multiplication import CModMulK
+from qualtran.resource_counting import BloqCountDictT, SympySymbolAllocator
+from qualtran.symbolics.types import is_symbolic, SymbolicInt
+
+
+@frozen
+class RSAPhaseEstimate(Bloq):
+    """Perform a single phase estimation of ModExp for the given base.
+
+    Computes the phase estimation of a single run of Modular Exponentiation with
+    an optional, pre-set base or a random, valid base.
```

Comment:
be more precise here. "The constructor requires a pre-set base, see the `make_for_shor` factory method for picking a random, valid base"